### PR TITLE
Add support for foundationdb in integration test suite

### PR DESCRIPTION
**Pull request by viveksinghggits**
## Change Overview
This change adds foundationdb in our integration test suite.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [x] :robot: Test

## Issues

- #452 

## Test Plan
To test the foundationdb we can use below command 
```
make integration-test
```
The logs can be found at [this gist](https://gist.github.com/viveksinghggits/b5b60597bb75975067e0b77fa20333ad).

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -65,6 +65,7 @@ github.com/elazarl/goproxy v0.0.0-20190711103511-473e67f1d7d2/go.mod h1:/Zj4wYkg
 github.com/elazarl/goproxy/ext v0.0.0-20190711103511-473e67f1d7d2/go.mod h1:gNh8nYJoAm43RfaxurUnxr+N1PwuFV3ZMl/efxlIlY8=
 github.com/emicklei/go-restful v0.0.0-20170410110728-ff4f55a20633/go.mod h1:otzb+WCGbkyDHkqmQmT5YD2WR4BBwUdeQoFo8l/7tVs=
 github.com/emicklei/go-restful v2.11.1+incompatible/go.mod h1:otzb+WCGbkyDHkqmQmT5YD2WR4BBwUdeQoFo8l/7tVs=
+github.com/evanphx/json-patch v4.5.0+incompatible h1:ouOWdg56aJriqS0huScTkVXPC5IcNrDCXZ6OoTAWu7M=
```
Comment (tdmanv) [inline go.sum]: the go mod changes are unrelated. do we need to merge these separately or are they spurious?

Comment (viveksinghggits) [reply go.sum]: Hi @tdmanv 
I am checking that just give me a second.

**Comment by viveksinghggits** [comment]
> this looks great! Please include the output when running the test in the PR description

Hi @tdmanv 
I actually mentioned the link to the log of the test. that is in a secret gist.

File: go.sum (hunk 1 above)
Comment (viveksinghggits) [reply go.sum]: I have made the corrections, the change in the go.sum file was unnecessary.
